CONTRIBUTING.md

```diff
@@ -0,0 +1,146 @@
+
+## Contribute Changes
+
+Here's the standard workflow to contribute changes to Eclipse-BaSyx-Python.
+
+Before contributing, please make sure, you fill out
+the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php). This
+is done by creating an Eclipse account for your git e-mail address and then submitting
+the following
+form: [https://accounts.eclipse.org/user/eca](https://accounts.eclipse.org/user/eca).
+The E-Mail address used to sign the ECA is the same one that needs to be used for
+committing.
+
+After this, the workflow to submit contributions to Eclipse-BaSyx-Python is pretty
+standard, as the picture (based
+on [this blog-post by Tomas Beuzen](https://www.tomasbeuzen.com/post/git-fork-branch-pull/))
+below shows:
+
+![CONTRIBUTING_Workflow](./etc/CONTRIBUTING_Workflow.png)
+
+1. Fork the Eclipse-BaSyx Repository
+2. Clone your fork to your development machine and add Eclipse-BaSyx as `upstream`:
+
+```bash
+git remote add upstream https://github.com/eclipse-basyx/basyx-python-sdk
+```
+
+3. Pull the branch you want to contribute to:
+
+```bash
+git pull upstream <branch_name>
+```
+
+Now, you can create a new local branch in which you can create your changes and actually
+do your changes. When you're done with that, continue with:
+
+4. Push the new branch to your fork:
+
+```bash
+git push origin <your_new_branch>
+```
+
+5. Create a [Pull Request](https://github.com/eclipse-basyx/basyx-python-sdk/pulls) from
+   your fork `<your_new_branch>` to the Eclipse-BaSyx-Python `<branch_name>`
+
+The Eclipse-BaSyx-Python maintainers will then review the pull request and communicate
+the further steps via the comments.
+
+## Commit and Pull Request Messages
+
+In order to effectively communicate, there's some conventions to respect when writing
+commit messages and pull requests.
+
+Similarily, to when creating an issue, the commit title, as well as the PR title should
+be as short as possible, ideally 72 characters or less. In order to achieve this, use
+imperative. Furthermore, if a specific file is affected by whatever the issue is about,
+please mention it at the beginning of the issue title.
+
+Here are some examples:
+
+```
+model.datatypes: Add type `xs:someThingMadeUp`
+compliance_tool: Fix fail to check aasx package without thumbnail
+adapter.aasx: Fix `Property.value` `0` converted into `NoneType`
+```
+
+The following guidelines are for the commit or PR message text:
+
+- No imperative, full text, bullet points where necessary
+- Max. 72 characters per line
+- There should be always 2 things in a Commit/PR message:
+    - Currently, the situation is this
+    - Motivate, why is it now different?
+- Don't describe what has been done, as this can be looked up in the code
+- Write as long as necessary, as short as possible
+- Where sensible, reference the specification, ideally
+  via `https://link/to.pdf#Page=123`
+- Optionally, where applicable reference respective issues: `Fixes #123`

```

Comment:
"Fixes" isn't imperative, maybe "Fix" should be used instead?

Reply:
GitHub recognizes both as command to close the issue, if the respective PR is merged. Since we only ask for imperative in the titles, I think it is valid to use here.